ios/MaasNearbyTransit/Sources/MaasNearbyTransit/Components/GradientBoundsScrollView.swift

```diff
@@ -0,0 +1,149 @@
+struct ScrollViewData: Equatable {
+    let size: CGSize
+    let contentOffset: CGFloat
+    let contentSize: CGSize
+
+    static var empty: Self { .init(size: .zero, contentOffset: .zero, contentSize: .zero) }
+}
+
+public struct GradientBoundsScrollView<Content>: View where Content: View {
+
+    private let axis: Axis.Set
+    private let showsIndicators: Bool
+    private let gradientColors: [Color]?
+    private let gradientLenght: CGFloat
+    private let content: Content
+
+    @State private var scrollData: ScrollViewData = .empty
+    @Environment(\.uiColorBackground) var backgroundColor
+
+    public init(
+        axis: Axis.Set = .horizontal,
+        showsIndicators: Bool = true,
+        gradientColors: [Color]? = nil,
+        gradientLenght: CGFloat = 20,
+        @ViewBuilder content: () -> Content
+    ) {
+        self.axis = axis
+        self.showsIndicators = showsIndicators
+        self.gradientColors = gradientColors
+        self.gradientLenght = gradientLenght
+        self.content = content()
```

Comment:
Interesting that this is only called on `init` - can the content be updated?

Reply:
Yes, we can update `content()` but `GradientBoundsScrollView` is a wrapper that applies scrolling behaviour on received content. So basically we'll make any updates that is necessary in place where we creating content.